### PR TITLE
Bug 1142147 Content Services pages

### PR DESCRIPTION
**Pull request by sgarrity**
Still in progress, but would like initial review on the implementation of the form at /contentservices/start/

It is based on the form at /about/partnerships/

CC: @pmclanahan

### CODE REVIEW

File: bedrock/mozorg/forms.py [hunk 1]
```diff
@@ -506,6 +509,280 @@ def __init__(self, *args, **kwargs):
         )
 
 
+class ContentServicesForm(forms.Form):
+    states = (
```
Comment (pmclanahan) [inline bedrock/mozorg/forms.py]: I don't see where this list of states is used.

Comment (sgarrity) [inline bedrock/mozorg/forms.py]: This was leftover cruft. Removed! Thanks.

**Comment by pmclanahan** [comment]
Sorry about taking forever on this @sgarrity. Things are looking good though. Do you still need me to help with the blank initial option for the state field?

**Comment by sgarrity** [comment]
> Sorry about taking forever on this @sgarrity. Things are looking good though. Do you still need me to help with the blank initial option for the state field?

No worries. We had been waiting on content in the meantime anyhow. Yes, if you can help with the blank initial option for the state field, that would be helpful.

**Comment by sgarrity** [comment]
@pmclanahan Also, when you select United States as the country, the Province field hides and the State field appears. This is working for me locally, but not on demo3. Any idea why that might be?

**Comment by pmclanahan** [comment]
@sgarrity it's because the output file name for the form JS wasn't changed and is being overwritten by the other "partnerships-bundle.js" file.

**Comment by pmclanahan** [comment]
not sure why it's working locally actually

File: bedrock/settings/base.py [hunk 2]
```diff
@@ -1138,6 +1146,16 @@ def JINJA_CONFIG():
...
         'output_filename': 'js/contact-spaces-ie7-bundle.js',
     },
+    'content-services-form': {
+        'source_filenames': (
+            'js/libs/jquery.validate.js',
+            'js/base/mozilla-form-helper.js',
+            'js/mozorg/content-services-form.js',
+            'js/base/mozilla-input-placeholder.js',
+        ),
+        'output_filename': 'js/partnerships-bundle.js',
```
Comment (pmclanahan) [inline bedrock/settings/base.py]: This should get a unique bundle file name.

File: bedrock/mozorg/views.py [hunk 3]
```diff
@@ -304,6 +305,88 @@ def partnerships(request):
...
+        msg = 'Form invalid'
+        stat = 400
+        success = False
+
+        if form.is_valid():
+            data = form.cleaned_data.copy()
+
+            honeypot = data.pop('office_fax')
+
+            if honeypot:
```
Comment (pmclanahan) [inline bedrock/mozorg/views.py]: Wouldn't returning an error for the spam trap just tell the spammers that there are spam traps they need to fix? I think it'd be better if we just made it seem like it succeeded and not submit anything to SF.

**Comment by sgarrity** [comment]
Ready for review!

**Comment by jgmize** [comment]
@sgarrity please rebase, thanks

**Comment by sgarrity** [comment]
> @sgarrity please rebase, thanks

Oops, the rebased version was in the mozilla/bedrock repo. I've pushed it here to my repo.

File: bedrock/mozorg/forms.py [hunk 4]
```diff
@@ -506,6 +510,238 @@ def __init__(self, *args, **kwargs):
...
+    )
+    phone = forms.CharField(
+        max_length=40,
+        required=True,
+        error_messages={
+            'required': _lazy(u'Please enter your phone number.')
+        },
+        widget=forms.TextInput(
+            attrs={
+                'size': 20,
```
Comment (jpetto) [inline bedrock/mozorg/forms.py]: Should add `required` attributes as with other fields?

File: bedrock/mozorg/templates/mozorg/contentservices/faq.html [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+      </p>
+    </div>
+  </section>
+
+  <section>
+    <h3 data-accordion-role="tab">{{ _('What do Tiles look like?') }}</h3>
+    <div data-accordion-role="tabpanel">
+      <p>
+        {{ _('A Tile is a visual bookmark that consists of one or two images: a required “static state” image and an optional (but recommended) “rollover state.”') }}
+        {% trans url='#TODO' %}
```
Comment (jpetto) [inline bedrock/mozorg/templates/mozorg/contentservices/faq.html]: Just calling out this `TODO`.

File: bedrock/mozorg/templates/mozorg/contentservices/faq.html [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+</section>
+
+<section id="requirements" class="accordion accordion-auto-init zebra">
+  <h2>{{ _('Requirements') }}</h2>
+
+  <section>
+    <h3 data-accordion-role="tab">{{ _('What are the technical requirements?') }}</h3>
+    <div data-accordion-role="tabpanel">
+      <p>
+      {% trans url='#TODO' %}
```
Comment (jpetto) [inline bedrock/mozorg/templates/mozorg/contentservices/faq.html]: Another `TODO`.

File: bedrock/mozorg/templates/mozorg/contentservices/about.html [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+{% extends "mozorg/contentservices/base.html" %}
+
+{% block page_title %}{{ _('About Content Services') }}{% endblock %}
+{% block body_id %}contentservices-about{% endblock %}
+{% block body_class %}sand{% endblock %}
+
+{% block article %}
+  <h1 class="title-shadow-box">{{ self.page_title() }}</h1>
+
+
```
Comment (jpetto) [inline bedrock/mozorg/templates/mozorg/contentservices/about.html]: Nit - extra line break.

File: bedrock/mozorg/templates/mozorg/contentservices/index.html [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+{% extends "mozorg/base-resp.html" %}
+
+{% block page_title %}{{ _('Content Services') }}{% endblock %}
+{% block body_id %}contentservices{% endblock %}
+{% block body_class %}sand{% endblock %}
+
+{% block site_css %}{% endblock %}
+
+{% block extrahead %}
+  {% stylesheet 'contentservices' %}
```
Comment (jpetto) [inline bedrock/mozorg/templates/mozorg/contentservices/index.html]: Is there a reason you're using the `extrahead` block here, but `page_css` in other files (like `start.html`)?

File: bedrock/mozorg/templates/mozorg/contentservices/tiles.html [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+  <p>
+  {{ _('After users browse, they generate their own browsing histories, and Firefox begins to use Tiles to show suggested sites, based on sites that the user has visited frequently and recently.') }}
+  {{ _('Mozilla Suggested Tiles offer sponsored placement in a designated interest category.') }}
+  {{ _('The browser makes these suggestions based on categories of sites that the user has already shown an interest in.') }}
+  {{ _('To protect users’ privacy, Mozilla strictly limits what data we collect about users’ interaction with Tiles.') }}
+  {{ _('We don’t collect browsing history, always encrypt data in transmission and discard IP addresses after a week.') }}
+  </p>
+
+  <p>
+  {{ high_res_img('img/contentservices/suggested-tiles.jpg', {'alt': 'Suggested Tiles screenshot', 'width': '620'}) }}
```
Comment (jpetto) [inline bedrock/mozorg/templates/mozorg/contentservices/tiles.html]: Nit - indentation is inconsistent in some of these `<p>` tags.

File: media/css/mozorg/contentservices.less [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+        }
+    }
+}
+
+.video {
+    .span(6);
+    float: right;
+    margin-top: -50px;
+    margin-bottom: @baseLine;
+    #welcome-video video {
```
Comment (jpetto) [inline media/css/mozorg/contentservices.less]: Nit - don't really need to nest an id selector (though it is nice for organizational purposes).

File: media/css/mozorg/contentservices.less [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+        .span(3);
+        list-style-type: none;
+        padding-top: 25px;
+        background: url(/media/img/mission/thin-stripe.png) center top repeat-x;
+    }
+
+    h4 {
+        margin-bottom: @baseLine / 6;
+        a:after {
+            content: " »";
```
Comment (jpetto) [inline media/css/mozorg/contentservices.less]: Use `trailing-arrow()` mixin?

File: media/js/mozorg/content-services-form.js [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+
+    var scrollup = function() {
+        $htmlBody.animate({ scrollTop: $mainContent.offset().top }, 500);
+    };
+
+    // If Campagin Type is set to "Other", then show the description field
+
+    var campaign_type_other = function() {
+        return $campaign_type.val().indexOf('Other') > -1;
+        // TODO: Is this an l10n issue?
```
Comment (jpetto) [inline media/js/mozorg/content-services-form.js]: Just pointing out this `TODO`.

File: media/js/mozorg/content-services-form.js [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+    var $campaign_type = $('#campaign_type');
+    var $typeFields = $('.type-field');
+    var $typeInputs = $typeFields.find('input, textarea, select');
+
+    var $country = $('#country');
+    var $stateField = $('.state-field');
+    var $stateFlydown = $('#state');
+    var $provinceField = $('.province-field');
+    var $stateInput = $stateField.find('input, textarea, select');
+    var $provinceInput = $provinceField.find('input, textarea, select');
```
Comment (jpetto) [inline media/js/mozorg/content-services-form.js]: Doesn't look like `$stateInput` or `$provinceInput` are used anywhere?

File: media/js/mozorg/content-services-form.js [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+    $campaign_type.on('change', function() {
+        toggleTypeFields(campaign_type_other());
+    });
+
+    // If Country is set to "United States", then hide Prov/Region field
+    // and show State field
+
+    var country_is_us = function() {
+        return $country.val().indexOf('us') > -1;
+    }
```
Comment (jpetto) [inline media/js/mozorg/content-services-form.js]: Missing semicolon above.

File: media/js/mozorg/content-services-form.js [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+                $stateField.val('');
+            }
+
+            $.ajax({
+                url: $sfForm.attr('action'),
+                data: $sfForm.serialize(),
+                type: $sfForm.attr('method'),
+                dataType: 'json',
+                success: function(data, status, xhr) {
+                    $('#sf-form').fadeOut('fast', function() {
```
Comment (jpetto) [inline media/js/mozorg/content-services-form.js]: Could use the `$sfFrom` variable declare at the top?

File: bedrock/mozorg/forms.py [hunk 16]
```diff
@@ -506,6 +510,238 @@ def __init__(self, *args, **kwargs):
...
+        locale = kwargs.get('locale', 'en-US')
+        country_list = product_details.get_regions(locale).items()
+        country_list = sorted(country_list, key=lambda country: country[1])
+        country_list.insert(0, ('', ''))
+        self.fields['country'].choices = country_list
+
+    def clean(self):
+        data = super(ContentServicesForm, self).clean()
+        if data.get('country') == 'us' and not data.get('state'):
+            raise ValidationError(self.fields['state'].error_messages['invalid'])
```
Comment (jpetto) [inline bedrock/mozorg/forms.py]: If JS is disabled (I know, I know) and the user is in the US, there is no way to avoid this error.

Comment (sgarrity) [inline bedrock/mozorg/forms.py]: I've updated the field to only be hidden if JS is enabled.

File: media/js/mozorg/content-services-form.js (hunk 12 above)
Comment (sgarrity) [inline media/js/mozorg/content-services-form.js]: This isn't a blocker for launch, but should be addressed before the page is localized.

**Comment by sgarrity** [comment]
@jpetto Thanks for the thorough review. I think I have everything addressed.

**Comment by jpetto** [comment]
Things are looking good. Just needs a squash. r+ :octopus:

**Comment by sgarrity** [comment]
Thanks @jpetto 

@pmclanahan How do you recommend I handle a squash since we both have commits in there?

**Comment by sgarrity** [comment]
Squashed!

All of my commits and squashed into one commit, and all of @pmclanahan's into another commit.

@pmclanahan: I took some liberties combining your commit messages into a single commit message. Let me know if there are any issues there.

**Pull request merged**